src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -4380,7 +4389,16 @@ public static void Sort<TKey, TValue, TComparer>(this Span<TKey> keys, Span<TVal
 
             if (keys.Length > 1)
             {
-                ArraySortHelper<TKey, TValue>.Default.Sort(keys, items, comparer); // value-type comparer will be boxed
+                if (typeof(TComparer).IsValueType)
```

Comment:
What's the benefit of special-casing valuetypes vs. references on this path?

Reply:
I meant whether this can be just an unconditional call to `ArraySortHelperFor<TKey, TValue, TComparer>.Sort(keys, items, comparer);`? 

Are there any cases that are faster thanks to `if (typeof(TComparer).IsValueType)` special casing?